requirements/requirements.in

```diff
@@ -47,6 +48,7 @@ schema
 suds-jurko==0.6
 tropo-webapi-python==0.1.3
 text-unidecode>=1.2
+typing-extensions  # Can be removed after upgrading to Python 3.8
```

Comment:
Can we get our [type annotation automation](https://docs.google.com/document/d/1SxDhlyBJVBcn1Jc2GHtIXyDtv2BTBPxiIxoks0lhXGk/edit) working before adding this?

Reply:
I played with **pytype** and **MyPy** briefly, and could not see how to get either of them to run without a long start-up time. I could just be missing something though. **MyPy** looks like it might also take quite a bit of effort to configure to suit our needs. I suspect we will be on Python 3.8 long before we get type annotation automation working.

Until then I am thinking of type annotation as a form of documentation, with all the pros and cons that documentation brings with it. As you can see, the `BackoffDetails` class is just to show what will be passed to handlers; I'm not using it for anything else. I considered just copy/pasting from [the backoff docs](https://github.com/litl/backoff#event-handlers), but thought that would have documentation's cons without the pros of type hints. I strongly considered just having a URL in a comment: Less likely to get out-of-date, but also less useful, because you first have to open the URL before you get any benefit from it. Maybe still a better option though.

If type annotation automation isn't going to happen any time soon, maybe we should revisit that conversation. I think our team understands the benefits and disadvantages of documentation; we know when we should add it, and when we should remove it. We continue to write it, despite having no tools to check when it gets stale. Without tools to do the same for type annotations, I think our documentation instinct is what we need for type annotations.

Reply:
(Added sample pre-commit hook for (local) type annotation automation to draft [guideline doc](https://docs.google.com/document/d/1SxDhlyBJVBcn1Jc2GHtIXyDtv2BTBPxiIxoks0lhXGk/edit#heading=h.ihmtncn4m2mn).)